### PR TITLE
`lgdo.Table.eval()` change to use NumExpr2

### PR DESCRIPTION
**Pull request by patgo25**
Changed `LGDO::Table.eval()` from using pandas eval() function to using directly the NumExpr module.
For two reasons:
1.  `LGDO::Table` -> `pandas data frame' ->` LGDO::Table` seems inefficient to me
2. The "old" code can't evaluate `ArrayOfEqualSizedArrays` .

Further improvements:
- JSON syntax is more like DSP syntax: `expression:  "@a+@b*var" `--> `expression:"a+b*var"`

Changes to 'test_table_eval.py`:
- Included test form 'ArrayOfEqualSizedArrays' evaluations (`ArrayOfEqualSizedArrays `-> `ArrayOfEqualSizedArrays `and `ArrayOfEqualSizedArrays `-> `Array  `(i.e sum(var, axis=1)) manipulations.
- Extended tests to test logical operations

Change to `test_build_hit.py`:
- Adjusted the test JSON configs to the new format

Solves issue #362

### CODE REVIEW

File: src/pygama/lgdo/table.py [hunk 1]
```diff
@@ -243,33 +247,70 @@ def eval(self, expr_config: dict) -> Table:
...
+                    raise TypeError("Data of type VectorOfVectors not supported (yet)")
+
+            out_data = ne.evaluate(
+                f"{spec['expression']}",
+                local_dict=dict(in_vars, **spec["parameters"])
+                if "parameters" in spec
+                else in_vars,
+                global_dict=None,
+                optimization="moderate",  # Slow, but calculation is accurate (alternative "aggressive")
+                truediv="auto",
```
Comment (gipert) [inline src/pygama/lgdo/table.py:290]: I would remove these (i.e. default behavior), I don't think we care. My intuition is that the default `optimization='aggressive'` is fine, we are not doing precision calculations (or non-trivial floating-point operations).

**Review by gipert — COMMENTED** [review]
Like this very much, thanks a lot for the work @patgo25. Can you briefly report about this at the analysis call today? We can proceed with merging then.

File: src/pygama/lgdo/table.py (hunk 1 above)
Comment (patgo25) [reply src/pygama/lgdo/table.py]: Done
i.e optimization is now 'aggressive' (default option) --> faster but small errors can occur

**Review by gipert — APPROVED** [review]

**Comment by jasondet** [comment]
I like this change. However I think the function signature should be a bit more apparent, right now all of the functionality is hidden in the various ways one can make an entry in expr_config. So one can't guess how to use it just by looking at the args. I would prefer something more like the following:

`tbl.eval(expr=None, pars_dict=None, out_name=None, append_to_self=False, return_nda=False, expr_dict=None)`

This matches more closely the signature of ne.evaluate(). If out_name is None you get an lgdo back (Array or ArrayOfEqualSizedArrays), unless return_nda=True in which case one just gets the ndarray returned by ne.evaluate. If out_name is not None, you get a table back with the column named by out_name; if append_to_self is true the column gets added to the table being evaluated. Then if expr_dict is not None you get something similar to the present behavior, and it will call itself recursively and build the table to return, or append to self according to the append_to_self option.

**Comment by patgo25** [comment]
Hi Jason, I didn't touch the declaration of `tbl.eval()`, I am not sure where all in pygama `tbl.eval()` is referenced (which would need to be changed accordingly), and if I understood your envisioned changes correctly. If you have some time, I would propose you change this in a separate pull request since in my opinion, this is a more significant change beyond the scope of this PR.

**Comment by jasondet** [comment]
That's a good point @patgo25. Let's proceed as you recommended. I'll sign off on it and merge it as-is.

**Review by jasondet — APPROVED** [review]

**Pull request merged**